tests/WebAppCallsMicrosoftGraph/WebSignInCredential.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.Graph;
+using Microsoft.Identity.Web;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace WebAppCallsMicrosoftGraph
+{
+    public class WebSignInCredential : IAuthenticationProvider
```

Comment:
@navyasric , as discussed
This would require Microsoft.Identity.Web to depend on the Graph SDK.
What do you think, @jennyf19 @pmaytak @henrik-me @bgavrilMS ?

I would think that have another NuGet package Microsoft.Identity.Web.MsGraph that would depend on both Microsoft.Identity.Web and Graph SDK.

Reply:
I have a query regarding the use of National clouds through Graph SDK.
By default, the configured endpoint uses MS graph global service i.e., graph.microsoft.com. To use other national cloud services, do we need to pass the specific URL or do we have any method to handle this?

Reply:
Actually this is a good point. I had thought of it yesterday, and then forgot. The AddMicrosoftGraph method should (optionally) take a base URL for Microsoft Graph

Reply:
Although creating a whole separate Nuget package seems like too much, I think it makes sense in this case to put this class and also the Graph service extensions into a separate package. Not every user will want to use Graph, so it'll keep the main package clean, plus maybe we might end up putting more Graph related code in there anyway.

Reply:
@jmprieur We reviewed this proposal and discussed with the working team today and I'm channeling some feedback here. We would like to avoid Graph specific dependencies and separate package. We want to design for Microsoft.Identity.Web to work with multiple APIs and have that flexibility.

Reply:
I'm under the impression that there exists an URI endpoint that you can query to find out national resources - AAD authority, Graph endpoint and others. Using a web service to discover endpoints is better than a package because it's easier to update values.

Reply:
@bgavrilMS : I don't think it exists yet for 3P.

Reply:
@pmaytak : I'm completely confused. I read one thing and its contrary? What is your recommendation? it's too much? or not?
> Although creating a whole separate Nuget package seems like too much, I think it makes sense in this case to put this class and also the Graph service extensions into a separate package. Not every user will want to use Graph, so it'll keep the main package clean, plus maybe we might end up putting more Graph related code in there anyway.

Reply:
@jmprieur   I wasn't sure of we could convince Azure.Identity to take a dependency on Microsoft.Identity.Web and then implement the WebSignInCredential there.  However, the .net 3.1 dependency is a problem to make that happen.
Alternatively, Microsoft.Identity.Web could take a dependency on Azure.Core for the TokenCredential class and implement the WebSignInCredential there.  Microsoft.Identity.Web would not need to depend on the Graph SDK.

@bgavrilMS You can get the Graph endpoint from the OpenID Connect discovery URL for some sovereign clouds.  If you look here https://login.microsoftonline.com/microsoft.onmicrosoft.com/v2.0/.well-known/openid-configuration  it is a property called msgraph_host